### PR TITLE
#114: Add tags to channels + refactoring (WIP)

### PR DESCRIPTION
**Pull request by 2Fast2BCn**

### CODE REVIEW

File: internal/channel_history/channel_history.go [hunk 1]
```diff
@@ -57,52 +56,38 @@ func getChannelHistory(db *sqlx.DB, chanIds []string, from time.Time,
...
-		getAll = true
-	}
-
-	qs, args, err := sqlx.In(sql, from, to, getAll, from, to, chanIds, from, to, chanIds)
+	rows, err := db.Queryx(sql, from, to, all, pq.Array(channelIds), commons.GetSettings().PreferredTimeZone)
 	if err != nil {
-		return r, errors.Wrapf(err, "sqlx.In(%s, %v, %v, %v, %v, %v, %v, %v, %v, %v, %v)",
-			sql, from, to, getAll, from, to, chanIds, from, to, chanIds)
+		return r, errors.Wrapf(err, "db.Queryx(%s, %v, %v, %v, %v, %v)",
+			sql, from, to, all, pq.Array(channelIds), commons.GetSettings().PreferredTimeZone)
```
Comment (ohenrik) [inline internal/channel_history/channel_history.go:89]: I think we have decided to stop adding variables to error messages, that is just my old style of error messages.

File: internal/channel_history/handlers.go [hunk 2]
```diff
@@ -78,25 +84,43 @@ func getChannelHistoryHandler(c *gin.Context, db *sqlx.DB) {
 		return
 	}
 
-	chanIds := strings.Split(c.Param("chanIds"), ",")
+	lndShortChannelIdStrings := strings.Split(c.Param("chanIds"), ",")
```
Comment (ohenrik) [inline internal/channel_history/handlers.go:87]: Would it be better to switch this to the Bolt spec version of the short id? We don't need to add that change to this, just wanted to mention it. the LND format has started to annoy me.

File: internal/channel_history/rebalancing_history.go [hunk 3]
```diff
@@ -54,49 +58,105 @@ func getRebalancingCost(db *sqlx.DB, from time.Time, to time.Time) (cost Rebalan
...
+//		return cost, nil
+//	}
+//
+//	if err != nil {
+//		return cost, err
+//	}
+//
+//	return cost, nil
+//
+//}
```
Comment (ohenrik) [inline internal/channel_history/rebalancing_history.go:113]: Forgot to delete this or is it temporary?

File: internal/channel_tags/database.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+//		  tag_origin_id,
+//		  tag_id,
+//		  created_on
+//		) values ($1, $2, $3, $4, $5, $6)
+//		RETURNING channel_tag_id;`,
+//		ct.FromNodeId, ct.ToNodeId, ct.ChannelId, ct.TagOriginId, ct.TagId, ct.CreatedOn).Scan(&ct.ChannelTagId)
+//	if err != nil {
+//		return channelTag{}, errors.Wrap(err, database.SqlExecutionError)
+//	}
+//
```
Comment (ohenrik) [inline internal/channel_tags/database.go:47]: Same here

File: internal/channels/database.go [hunk 5]
```diff
@@ -2,38 +2,108 @@ package channels
...
+	return channelId, nil
+}
+
+func addChannel(db *sqlx.DB, channel Channel) (Channel, error) {
+	channel.CreatedOn = time.Now().UTC()
+	channel.UpdateOn.Time = channel.CreatedOn
+	err := db.QueryRowx(`
+		INSERT INTO channel (
+		  short_channel_id,
+		  lnd_channel_point,
```
Comment (ohenrik) [inline internal/channels/database.go:93]: Is the channel point specific to LND? Or is it also in the Bolt spec? Regardless it's very handy to have as it's the open TX hash and the output index of the channel. 

That also makes me think if we should store the closing channel hash

**Review by ohenrik — COMMENTED** [review]
I'm not done yet, but it looks really good so far :)

File: internal/channel_history/channel_balance.go [hunk 6]
```diff
@@ -41,35 +52,35 @@ func getChannelBalance(db *sqlx.DB, lndShortChannelId string, from time.Time, to
 				(select time,
 				   -outgoing_amount_msat as amt
 				from forward
-				where lnd_outgoing_short_channel_id = ((table lnd_short_channel_id)::numeric)
+				where incoming_channel_id = $1
```
[the remark this answers is not in the document]
Comment (2Fast2BCn) [reply internal/channel_history/channel_balance.go:55]: damn nice, well spotted!

File: internal/channel_history/handlers.go (hunk 2 above)
Comment (2Fast2BCn) [reply internal/channel_history/handlers.go:87]: We should use our internal channelId everywhere ;-) no more bolt/cln/lnd stuff ;-)

File: internal/channels/database.go (hunk 5 above)
Comment (2Fast2BCn) [reply internal/channels/database.go:93]: replaced lnd_channel_point with funding_transaction_hash and funding_output_index
Also added closing_transaction_hash

File: internal/channel_history/handlers.go (hunk 2 above)
Comment (ohenrik) [reply internal/channel_history/handlers.go:87]: Ah ofc. I forgot this was our own channel ids (torq channel ID)

**Pull request merged**